tcms/testruns/static/testruns/js/get.js

```diff
@@ -277,6 +277,15 @@ function fileBugFromExecution(execution) {
         const trackerId = $('.one-click-bug-report-form #id-issue-tracker').val()
         jsonRPC('Bug.report', [execution.id, trackerId], result => {
 
+            const testExecutionRow = $(`.test-execution-${execution.id}`)
+            animate(testExecutionRow, () => {
+                testExecutionRow.find('.test-execution-hyperlinks').append(renderLink({ is_defect: true, url: result.response }))
+
+                const bugCountEl = testExecutionRow.find('.test-execution-bugs-count')
+                const bugCount = parseInt(bugCountEl[0].innerText)
```

Comment:
we already do this in 2 places. If we start doing it more ofter we can instead start keeping state in JS. then this will boil down to:
```javascript
testExecutions[testExecution.id].bugCount++
testExecutionRow.find('.test-execution-bugs-count').html(testExecutions[testExecution.id].bugCount)
```
then at some point we will get tired of reinventing the wheel, migrate to a proper JS framework and the code will be:
```javascript
testExecutions[testExecution.id].bugCount++
// framework handles DOM mutations based on state change
```

Also, we can omit `parseInt` here and still get the same result, because of all the stuff JS is doing under the hood, converting types, etc. but I prefer to be explicit and fail here, instead of silently getting some weird results in the UI

Reply:
+1 for keeping the state. It is already required elsewhere and the data is available anyway unless the user refreshes. 

> then at some point we will get tired of reinventing the wheel,

We aren't reinventing, rather taking the evolutionary path to where using the wheel will make natural sense.

parseInt() is good, better be explicit than not.

Reply:
> +1 for keeping the state.

let's do it in another PR, there are more places to be cleanup and it will be faster if I do it all at once.

> We aren't reinventing, rather taking the evolutionary path to where using the wheel will make natural sense.

I suppose you are right, I just feel like reinventing the wheel every time I spend 30 mins troubleshooting a problem like this.

Reply:
Added as item 15 in your backlog. +1 for dealing with this at a later stage, possibly even after the initial release of this redesign.